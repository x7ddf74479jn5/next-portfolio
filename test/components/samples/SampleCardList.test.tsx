import { render, screen } from "@testing-library/react";
import SampleCardList from "src/components/samples/SampleCardList";

describe("SampleCardList", () => {
  const testData = [
    {
      id: "id",
      title: "title",
      category: "category",
      description: "description",
      link: "link",
      imgSrc: "/",
      imgAlt: "alt",
    },
  ];

  it("matches snapshot", () => {
    const { asFragment } = render(<SampleCardList samples={testData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    render(<SampleCardList samples={testData} />);
    const listItemElements = screen.getAllByRole("listitem");
    expect(listItemElements.length).toBeGreaterThan(0);
  });
});
