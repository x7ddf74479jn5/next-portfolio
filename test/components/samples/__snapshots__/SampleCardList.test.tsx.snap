// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleCardList matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="listSamples"
  >
    <li>
      <div
        class="sampleCard"
      >
        <a
          class="testLineNone"
          href="/link"
        >
          <div
            class="mediaThumb"
          >
            <div
              style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
            >
              <noscript />
              <img
                alt="alt"
                data-nimg="true"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
            </div>
          </div>
          <div
            class="listSamplesDetail"
          >
            <h3
              class="title"
            >
              title
            </h3>
            <p
              class="category"
            >
              category
            </p>
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 5H11l5 7-5 7h4.5l5-7z"
                />
                <path
                  d="M8.5 5H4l5 7-5 7h4.5l5-7z"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
